enclave-runner/src/usercalls/mod.rs

```diff
@@ -719,9 +720,14 @@ impl EnclaveState {
                                         panic!("{}", &panic);
                                     }
                                     Err(EnclaveAbort::Exit{ panic })
-
                                 }
-                                Err(EnclaveAbort::Exit { panic: false }) => Ok((0, 0)),
+                                Err(EnclaveAbort::Exit { panic: false }) => match state.mode {
+                                    EnclaveEntry::ExecutableMain => Ok((0, 0)),
+                                    EnclaveEntry::Library |
+                                    EnclaveEntry::ExecutableNonMain => Err(EnclaveAbort::Exit{
+                                        panic: EnclavePanic::DebugStr("Execution aborted by a thread".to_owned()),
```

Comment:
We should probably change this so that calling `std::process::exit(0)` from a secondary thread is similar to the main thread doing the same (i.e. not generating a panic). What do you think @jethrogb?

Reply:
I think any non main thread exiting is a violation of the sgx spec

Reply:
I don't think it is. Do you have a reference?

Reply:
In fact this is explicitly allowed (because Rust allows it)

Reply:
I have already addressed this in b8a65d89e72f8922210fe6d0ec5ffe817c44cccc. @jethrogb could you do a final review?